pages/api/update-blogs.ts

```diff
@@ -0,0 +1,36 @@
+import Crypto from 'crypto';
+
+import Boom from '@hapi/boom';
+import { string, object } from 'yup';
+
+import { withAsync, withValidation } from '../../api-helpers/api-hofs';
+import { getConfig } from '../../api-helpers/config';
+import { updateBlogs } from '../../api-helpers/feedFunctions';
+
+export default withAsync(
+  withValidation({
+    body: object({ secret: string().required() }).required(),
+  })(async (req) => {
+    if (req.method !== 'PATCH') {
+      throw Boom.notFound();
+    }
+
+    try {
+      const isSecretValid = Crypto.timingSafeEqual(
+        Buffer.from(req.body.secret),
+        Buffer.from(getConfig('FEED_UPDATE_SECRET')),
+      );
+
+      if (!isSecretValid) {
+        console.log('not valid secret');
+        throw Boom.unauthorized();
+      }
+    } catch (err) {
+      throw Boom.isBoom(err) ? err : Boom.unauthorized();
+    }
+
+    await updateBlogs();
+
+    return null;
+  }),
+);
```

Comment:
Ile czasu to zajmuje dla wszystkich blogów aktualnie?

Reply:
~10 sekund